### PR TITLE
Fix JSON schema for zarf.yaml

### LINKED ISSUE #925: JSON Schema oneOf for zarf chart is incorrect
The generated schema is not currently defining the oneOf property for the charts array properly leading to incorrect parsing when using the JSON schema file.

### PR DESCRIPTION
**Pull request by corang**
## Description

Type annotation changed for zarfChart fields to create a proper "oneOf" schema.

## Related Issue

Fixes #925 

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## Checklist before merging

- [X] Documentation has been updated as necessary (add the `needs-docs` label)

### CODE REVIEW

**Comment by jeff-mccoy** [comment]
I noticed this issue the other day too, can you provide a screenshot (maybe before/after) of the issue resolved in vscode?

File: src/types/component.go [hunk 1]
```diff
@@ -73,12 +73,12 @@ type ZarfFile struct {
 type ZarfChart struct {
 	Name        string   `json:"name" jsonschema:"description=The name of the chart to deploy, this should be the name of the chart as it is installed in the helm repo"`
 	ReleaseName string   `json:"releaseName,omitempty" jsonschema:"description=The name of the release to create, defaults to the name of the chart"`
-	Url         string   `json:"url,omitempty" jsonschema:"oneof_required=chartLocationGroup,url,description=The URL of the chart repository or git url if the chart is using a git repo instead of helm repo"`
+	Url         string   `json:"url,omitempty" jsonschema:"oneof_required=url,description=The URL of the chart repository or git url if the chart is using a git repo instead of helm repo"`
```
[the remark this answers is not in the document]
Comment (corang) [reply src/types/component.go:76]: I'm almost positive this isn't how to do it in this case. With the code as is this json is generated:
```json
      "oneOf": [
        {
          "required": [
            "url"
          ],
          "title": "url"
        },
        {
          "required": [
            "localPath"
          ],
          "title": "localPath"
        }
      ]
```
With the suggestion here the json generated is:
```json
      "oneOf": [
        {
          "required": [
            "url",
            "localPath"
          ],
          "title": "chartLocationGroup"
        }
      ]
```
Given how we want it to be either `url` or `localPath` having them in the same object under oneOf isn't correct according to the JSON schema docs https://json-schema.org/understanding-json-schema/reference/combining.html#oneof

[the remark this answers is not in the document]
Comment (corang) [reply src/types/component.go:76]: Without the chart group (as is):
image
With the chart group added (proposed change):
image

[the remark this answers is not in the document]
Comment (jeff-mccoy) [reply src/types/component.go:76]: wait I thought we wanted it to allow localPath or url? are these screenshots mislabled?

[the remark this answers is not in the document]
Comment (Racer159) [reply src/types/component.go:76]: @corang you are correct actually, I misunderstood it, but here is the upstream fixture: https://github.com/alecthomas/jsonschema/blob/cc89723c9db01ad5b2b9b5b718432ee103d12197/fixtures/oneof.json

**Review by Racer159 — APPROVED** [review]

**Comment by corang** [comment]
@jeff-mccoy 
image
image
Differences are the branch that my settings are referring to `master` vs `925-schema-error` and whether or not there is a red squiggle

**Comment by corang** [comment]
Error pop-up
image

**Pull request merged**